lib/matplotlib/axes/_base.py

```diff
@@ -2949,15 +2955,19 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
             left = kw.pop('xmin')
         if 'xmax' in kw:
             right = kw.pop('xmax')
+        # _converter is private, usually True, but can be false
+        _converter = kw.pop('_converter', True)
         if kw:
             raise ValueError("unrecognized kwargs: %s" % list(kw))
 
         if right is None and iterable(left):
             left, right = left
 
-        self._process_unit_info(xdata=(left, right))
-        left = self._validate_converted_limits(left, self.convert_xunits)
-        right = self._validate_converted_limits(right, self.convert_xunits)
+        if _converter:
```

Comment:
What's the purpose of this if statement?

Reply:
As above.  Its because `cla` needs the `set_x/ylim` call.  I tried getting rid of it, but bad things seemed to happen.